modules/currency-l0/src/main/scala/org/tessellation/currency/infrastructure/healthcheck/HealthCheckDaemon.scala

```diff
@@ -0,0 +1,34 @@
+package org.tessellation.currency.infrastructure.healthcheck
+
+import cats.effect.Async
+import cats.effect.std.Supervisor
+import cats.syntax.functor._
+
+import scala.concurrent.duration._
+
+import org.tessellation.currency.modules.HealthChecks
+import org.tessellation.sdk.domain.Daemon
+
+import fs2._
+
+trait HealthCheckDaemon[F[_]] extends Daemon[F] {}
+
+object HealthCheckDaemon {
+
+  def make[F[_]: Async](healthChecks: HealthChecks[F])(implicit S: Supervisor[F]): HealthCheckDaemon[F] = new HealthCheckDaemon[F] {
```

Comment:
It can be extracted. I will create a common trait with `def trigger` method and just `healthChecks.trigger()`. Thanks to that Daemon does not care what kind of triggers we have and can be reused.